### ISSUE TITLE
Reproducible modelling

### ISSUE THREAD

**Issue by lamres** [awslabs/autogluon#1454]
Hi.
1. Is there any way to make a reproducible fit operation? Some kinda random seed or similar. E.g., getting the same result if I put the same data, time limit and other hyperparameters.
2. Is there any way to re-fit the trained model on new dataset?

**Comment by willsmithorg** [comment]
From what I've seen of the Autogluon code, anywhere a random number is used, the seed is always fixed (usually to 0).
This may not be the case with the downstream models, but from what I've seen downstream models also take a seed parameter.
I believe it's a desired feature that fit() is reproducible.
I suggest you post some code that reproduces what you see, with enough sample data to trigger this.

**Comment by Innixma** [comment]
AutoGluon is mostly reproducible by default as it fixes all seeds to 0 whenever possible. However, there are some exceptions:

1. When time limit is specified two runs could take different amounts of time to train identical models. This will lead one run to have more time to train a model leading to a different model and thus different results.
2. FastAI Neural Network is not seeded in v0.3.1 and is different every time. This is fixed in upcoming v0.4.
3. Memory limits can cause AutoGluon to adaptively change how models are trained to avoid going OOM. Two runs could have different amounts of available memory which could cause differences. This will only happen if using ~20%+ of system memory.

**Comment by lamres** [comment]
Hi.
I returned back to this question. Just updated the library to 0.4. I found in documentation the next:
`This seed is NOT used in the training of individual models, for that you need to explicitly set the corresponding seed hyperparameter (usually called ‘seed_value’) of each individual model.`

When I try to put the seed_value parameter I got the next message:
`Unexpected the keyword argument "seed_value"`

What is the best practice to make a reproducible model? First of all, I need it for FASTAI, RF, CAT models.

**Comment by Shuyi-bomi** [comment]
Hi is there a way to change the random seed since I don't want the same model?

**Comment by shah-nawaz-alam** [comment]
Hi, I am not able to reproduce results over multiple runs with same default parameters and data (Autogluon version 0.5.2). I am using "medium_quality" and relying on "holdout_frac=0.1" for tuning. Can anyone throw some light on how to reproduce the results? Thanks

**Comment by shah-nawaz-alam** [comment]
Thank you @smolix for quick response. I did meant getting identical results over multiple runs. Currently, I am facing very high variance in the test metrics over multiple runs, which signals that the training is not stable. Is there a way to set seed on this random training somehow to give fix result?  

Other details: I am not using custom models, so default ones only.

**Comment by Innixma** [comment]
@shah-nawaz-alam can you provide a colab notebook to reproduce the non-deterministic result?
